src/sources/kafka.rs

```diff
@@ -2002,12 +2002,15 @@ mod integration_test {
 
         // 5. Total number of events processed should equal the number sent
         let total = events1.len() + events2.len();
-        assert_ne!(
-            events1.len(),
-            0,
-            "First batch of events should be non-zero (increase KAFKA_SHUTDOWN_DELAY?)"
-        );
-        assert_ne!(events2.len(), 0, "Second batch of events should be non-zero (decrease KAFKA_SHUTDOWN_DELAY or increase KAFKA_SEND_COUNT?) ");
+        // TODO
```

Comment:
The image update seems to have resulted in the first consumer no longer receiving events before the shutdown so I commented out the per-consumer assertions for now.

cc/ @jches I'm sorry to bother, but since you added this test, I expect you might more easily be able to figure out what's happening here.

Reply:
Interesting..I wouldn't expect a different image to matter much; I think I used the bitnami kafka images when I was doing this testing locally. IIRC this test sets up two consumers consecutively, so the first one runs, shuts down, and then the second runs, to test the "drain acknowledgements before shutdown" part.

If the first consumer isn't getting any messages, you could allow it to run for a little bit longer - the default is 2s and can be adjusted in the test code a few lines above this, or by environment variables.

Another thing worth looking into is consumer group wait times - I don't recall the exact setting but there is a configurable waiting period where the brokers will wait for more consumers to join before the consumers can do their initial partition assignment. If that's set longer than the shutdown delay here it might explain this behavior.

Reply:
Ah it's `group.initial.rebalance.delay.ms` and looks like the default is 3s https://cwiki.apache.org/confluence/display/KAFKA/KIP-134%3A+Delay+initial+consumer+group+rebalance . Not sure if the old images had a different setting that made this work..
I'd try setting that lower (1s would probably be fine in this case) in the broker config, or set the shutdown delay to 4 or 5s

Reply:
That did it! Thank you!!